File: Client/HoneyClient/src/app/data/interface/order/Order.ts
import { OrderDetailsItem } from "../order-detail/OrderDetail";

export interface OrderItem {
  id: number;
  totalPrice: number; // Цена продукта
  quantity: number; // Количество товара в корзине
  fullName: string; // Имя пользователя
  number: string; // Номер телефона
  status: string; // Статус заказа (например, Pending, Completed, Cancelled)
  orderDetails: OrderDetailsItem[]; // Связь с деталями заказа
}

File: Backend/HoneyShop.Domain/Model/Order.cs
namespace HonaeyShop.Domain.Model;

// Модель заказа
public class Order
{
    public int Id { get; set; } // Идентификатор заказа
    public int UserId { get; set; } // Идентификатор пользователя
    public User User { get; set; } // Связь с пользователем
    public decimal TotalPrice { get; set; } // Общая стоимость заказа
    
    public int StatusId { get; set; } // Идентификатор статуса заказа
    public OrderStatus Status { get; set; } // Статус заказа (например, Pending, Completed, Cancelled)
    public DateTime CreatedAt { get; set; } // Дата создания записи
    
    public string? Message { get; set; } // Сообщение о заказе
    
    public ICollection<History> Histories { get; set; } // Связь с историей изменений
    public ICollection<OrderDetail> OrderDetails { get; set; } // Связь с деталями заказа
}